scripts/check_phase44c_projection_root_probe.py

```
@@ -0,0 +1,287 @@
+
+
+def hash32_json(tag: str, value: Any) -> str:
+    payload = canonical_json({"tag": tag, "value": value})
+    return hash32_text(payload)
+
+
+def load_json(path: pathlib.Path) -> dict[str, Any]:
+    try:
+        data = json.loads(path.read_text(encoding="utf-8"))
+    except json.JSONDecodeError as exc:
+        raise Phase44CError(f"{path} is not valid JSON: {exc}") from exc
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)